--- a/dts/upstream/src/arm64/rockchip/rk356x-base.dtsi	2026-01-12 14:57:48.788862172 +0800
+++ b/dts/upstream/src/arm64/rockchip/rk356x-base.dtsi	2026-01-12 14:57:26.280260373 +0800
@@ -341,6 +341,12 @@
 		status = "disabled";
 	};
 
+	xpcs: syscon@fda00000 {
+		compatible = "rockchip,rk3568-xpcs", "syscon";
+		reg = <0x0 0xfda00000 0x0 0x200000>;
+		status = "disabled";
+	};
+
 	pmugrf: syscon@fdc20000 {
 		compatible = "rockchip,rk3568-pmugrf", "syscon", "simple-mfd";
 		reg = <0x0 0xfdc20000 0x0 0x10000>;
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -87,6 +87,28 @@ dtb-$(CONFIG_ROCKCHIP_RK3368) += \
        rk3368-geekbox.dtb \
        rk3368-px5-evb.dtb \

+dtb-$(CONFIG_ROCKCHIP_RK3399) += \
+       rk3399-guangmiao-g4c.dtb \
+       rk3399-nanopi-r4se.dtb
+
+dtb-$(CONFIG_ROCKCHIP_RK3566) += \
+       rk3566-nanopi-r3s.dtb \
+
+dtb-$(CONFIG_ROCKCHIP_RK3568) += \
+       rk3568-armsom-sige3.dtb \
+       rk3568-mmbox-anas3035.dtb \
+       rk3568-lyt-t68m.dtb \
+       rk3568-mrkaio-m68s.dtb \
+       rk3568-photonicat.dtb \
+       rk3568-x3568-v4.dtb \
+       rk3568-nsy-g68-plus.dtb \
+       rk3568-nsy-g16-plus.dtb \
+       rk3568-bdy-g18-pro.dtb \
+       rk3568-xiguapi-v3.dtb
+
+dtb-$(CONFIG_ROCKCHIP_RK3588) += \
+       rk3588-cyber3588-aib.dtb
+
 dtb-$(CONFIG_ARCH_S5P4418) += \
        s5p4418-nanopi2.dtb

